### PR TITLE
Update modules

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,14 +43,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#342](https://github.com/nf-core/taxprofiler/pull/342) Fixed docs/usage to correctly list the required database files for Bracken and tips to obtain Kraken2 databases (fix by @husensofteng)
 - [#350](https://github.com/nf-core/taxprofiler/pull/350) Reorganize the CI tests into separate profiles in preparation for implementation of nf-test (fix by @LilyAnderssonLee)
 - [#364](https://github.com/nf-core/taxprofiler/pull/364) Add autoMounts to apptainer profile in nextflow.config (♥ to @hkaspersento for reporting, fix by @LilyAnderssonLee)
+- [#372](https://github.com/nf-core/taxprofiler/pull/372/files) Update modules to use quay.io nf-core mirrored containers (♥ to @maxulysse for pointing out, fix by @LilyAnderssonLee and @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:46]:
```suggestion
- [#372](https://github.com/nf-core/taxprofiler/pull/372) Update modules to use quay.io nf-core mirrored containers (♥ to @maxulysse for pointing out, fix by @LilyAnderssonLee and @jfy133)
```

File: subworkflows/local/longread_hostremoval.nf [hunk 2]
```diff
@@ -48,7 +48,10 @@ workflow LONGREAD_HOSTREMOVAL {
     bam_bai = MINIMAP2_ALIGN.out.bam
         .join(SAMTOOLS_INDEX.out.bai)
 
-    SAMTOOLS_STATS ( bam_bai, reference )
+    // Convert the reference path into a channel
+    reference_channel = Channel.fromPath(reference)
+    SAMTOOLS_STATS(bam_bai, reference_channel.map { refPath -> [[:], refPath] })
```
Comment (jfy133) [inline subworkflows/local/longread_hostremoval.nf:53]:
```suggestion
    SAMTOOLS_STATS(bam_bai, reference_channel.map { refpath -> [[:], refpath] })
```

Convention :) groovy functions use camel case

File: subworkflows/local/shortread_hostremoval.nf [hunk 3]
```diff
@@ -30,13 +30,15 @@ workflow SHORTREAD_HOSTREMOVAL {
...
     ch_versions      = ch_versions.mix( SAMTOOLS_INDEX.out.versions.first() )
 
-    bam_bai = BOWTIE2_ALIGN.out.bam
+    bam_bai = BOWTIE2_ALIGN.out.aligned
         .join(SAMTOOLS_INDEX.out.bai, remainder: true)
 
-    SAMTOOLS_STATS ( bam_bai, reference )
+    // Convert the reference path into a channel
+    reference_channel = Channel.fromPath(reference)
+    SAMTOOLS_STATS(bam_bai, reference_channel.map { refPath -> [[:], refPath] })
```
Comment (jfy133) [inline subworkflows/local/shortread_hostremoval.nf:41]:
```suggestion
    SAMTOOLS_STATS(bam_bai, reference_channel.map { refpath -> [[:], refpath] })
```

**Review by jfy133 — APPROVED** [review]
LGTM if tests are passing and we get all the expected output! 

Only changes to check is that when converting `reference` to a channel in the two hostremoval workflows that the reference isn't being consumed after the first sample.

**Comment by LilyAnderssonLee** [comment]
> LGTM if tests are passing and we get all the expected output!
> 
> Only changes to check is that when converting `reference` to a channel in the two hostremoval workflows that the reference isn't being consumed after the first sample.

You are right. Only two samples were processed by SAMTOOLS_STATS by the new code.
    `reference_channel = Channel.fromPath(reference)`
    `SAMTOOLS_STATS(bam_bai, reference_channel.map { refPath -> [[:], refPath] })`

Outputs in the samtools/stats folder:
image

Here are the outputs based on original module. 
image

So we still need to fix this.

**Comment by LilyAnderssonLee** [comment]
The output of samtools/stats/ are correct now by using `reference_channel = reads.map { meta, _ -> file(reference) }`

**Comment by jfy133** [comment]
@LilyAnderssonLee I just ran the command based on https://github.com/nf-core/taxprofiler/pull/372/commits/34960d6b46410c64f31d28df4b0635d0becb38d2

And this works as expected for me? I do not get any error, and all the expected `stats` output files are there?

What command did you use to run?

```bash
 nextflow run ../main.nf -profile singularity,test_noprofiling --outdir ./results --input samplesheet_local.csv --databases database_v1.1_local.csv --run_diamond
```

**Comment by LilyAnderssonLee** [comment]
Yes, it also worked for me, and the output files were correct. The only concern was the use of underscore. I thought the use of 'underscore' is not so beautiful and try to find an alternative approach instead of using underscores.
The code I used was:
`nextflow run ../taxprofiler_dev/main.nf -profile test_noprofiling,singularity --outdir ./test_run_samtools_stats_new`

**Review by jfy133 — APPROVED** [review]
Samtools stats/view channel fixed and independently verified :+1:

**Pull request merged**